Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2381 @@
+validst3            = 'not-activated'
+
+/*********************************/
+/* Get program args              */
+/*********************************/
+parse arg argString
+if GetArgs(argString) <> 0 then
+  exit -1
+
+Select
+ When LPAR_Status = validst1 then NOP
+ When LPAR_Status = validst2 then NOP
+ When LPAR_Status = validst3 then NOP
+ OTHERWISE
+  do
+   call EntryArrow 
+   say 'Invalid LPAR status entered, default of operating will be used'
+   LPAR_Status = 'operating'
+   call ExitArrow 
+  end
+end
+
+/**************************************/
+/* Before proceeding, ensure the      */
+/* data set specified actually exists */
+/**************************************/
+if VerifyDataSet(TEMP_DATASET) <> 0 then
+  exit fatalErrorAndCleanup('** specified data set does not exist **')
+
+if localCPC then
+  say 'Obtaining LPARs on LOCAL CPC'
+else
+  say 'Starting query of LPARs on CPC:'||CPCname
+
+if ISVREXX then
+  do
+    say 'Running in an ISV REXX environment'
+    hwiHostRC = hwihost("ON")
+    say 'HWIHOST("ON") return code is :('||hwiHostRC||')'
+
+    if hwiHostRC <> 0 then
+      exit fatalErrorAndCleanup('** unable to turn on HWIHOST **')
+    HWIHOST_ON = TRUE
+  end
+
+call IncludeConstants
+
+call JSON_getToolkitConstants
+if RESULT <> 0 then
+  exit fatalErrorAndCleanup( '** Environment error **' )
+
+PROC_GLOBALS = 'VERBOSE parserHandle '||HWT_CONSTANTS
+
+/*********************************/
+/* Create a new parser instance. */
+/*********************************/
+parserHandle = ''
+call JSON_initParser
+if RESULT <> 0 then
+  exit fatalErrorAndCleanup( '** Parser init failure **' )
+
+/*********************************/
+/* Start of BCPii logic          */
+/*********************************/
+if localCPC then
+  do
+    if getLocalCPCInfo() <> 0 then
+      exit fatalErrorAndCleanup( '** failed to get LOCAL CPC info **' )
+  end
+else
+  do
+    if getCPCInfo() <> 0 then
+      do
+        errMsg = '** failed to get CPC info for ('||CPCname||') **'
+        exit fatalErrorAndCleanup(errMsg)
+      end
+    MEMBER_NAME = CPCname
+  end
+
+if getLPARList() <> 0 then
+  exit fatalErrorAndCleanup( '** failed to obtain list of LPARs **' )
+
+/****************************************
+  The CPC has at least one LPAR so prime
+  the list of attributes that need to be
+  retrieved for each LPAR
+****************************************/
+call PrepLPARAttributes
+call PrepHdrRow
+
+outLC = 1 /* HdrRow */
+do iLpar = 1 to LPARsList.0
+  outLC = outLC + 1
+  LPARuri = LPARsList.iLpar.uri
+  LPARtargetName = LPARsList.iLpar.targetName
+  LPARName = LPARsList.iLpar.Name
+  LPARstatus = LPARsList.iLpar.status
+  REXXWRT.outLC = LPARName||','||LPARstatus
+
+  /********************************************************
+   Only retrieve content for LPARs in the specified status
+   NOTE: Default is to retrive content for operating LPARs
+  *********************************************************/
+  if LPARstatus <> LPAR_Status then
```

Comment:
this is more for clarity, but i'd change one of the variables to be more descriptive..... looking at this, you can't tell what's being compared ...i know it's lpar status..but what is each one associated with? or is for?

Reply:
fixed